### PR TITLE
ref(ui) Convert more deprecated select components

### PR DESCRIPTION
**Pull request by markstory**
Move another batch of select controls to the new react-select version. I had to make AsyncControl aware of the deprecated mode to make it compatible with the new react-select implementation.

### CODE REVIEW

**Comment by priscilawebdev** [comment]
should we add a little spacing at the top of this selector?

image
image

**Comment by priscilawebdev** [comment]
We should also check this issue:

**Before:**
image

**After:**
image

File: src/sentry/static/sentry/app/components/forms/selectControl.tsx [hunk 1]
```diff
@@ -180,6 +180,7 @@ function SelectControl(props: ControlProps) {
       border: '1px solid #c2e0ff',
       display: 'flex',
       marginRight: '4px',
+      marginTop: '2px',
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/forms/selectControl.tsx:183]: maybe we could use` space(0.25)` here?

File: src/sentry/static/sentry/app/components/forms/selectControl.tsx [hunk 2]
```diff
@@ -180,6 +180,7 @@ function SelectControl(props: ControlProps) {
       border: '1px solid #c2e0ff',
       display: 'flex',
       marginRight: '4px',
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/forms/selectControl.tsx:182]: and `space(0.5)` here?

**Review by priscilawebdev — APPROVED** [review]
I left some suggestions but it looks good to me

**Comment by priscilawebdev** [comment]
it looks better, but I think we could add a little more spacing. Please see:

image

**Comment by markstory** [comment]
I chose to not fiddle with the margins too much as each change to make the multi line rendering better, made the single line state worse.

**Pull request merged**